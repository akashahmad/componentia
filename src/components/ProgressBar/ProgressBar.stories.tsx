import * as React from "react";

import type { Meta, StoryObj } from "@storybook/react";

import { ProgressBar } from "./ProgressBar";

const meta: Meta<typeof ProgressBar> = {
    component: ProgressBar,
    tags: ["autodocs"],
};

export default meta;
type Story = StoryObj<typeof ProgressBar>;

export const Default: Story = {
    args: {},
    argTypes: {
        color: {
            table: {
                type: { summary: "hex code, e.g #FFFFFF" },
                defaultValue: { summary: '"#25CBA1" ' },
            },
        },
    },
    render: props => <ProgressBar {...props} />,
};
