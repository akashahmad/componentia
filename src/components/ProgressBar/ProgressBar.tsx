import * as React from "react";

import cx from "classnames";

export function ProgressBar({ percentage = 50, color }: ProgressBarProps) {
    return (
        <div className={cx("bg-grey-10 w-full h-[7px] rounded-full overflow-hidden inline-block")}>
            <span
                style={{ width: `${percentage}%`, ...(color ? { background: color } : {}) }}
                className={cx("inline-block h-[7px] absolute rounded-full", {
                    ["bg-teal-light-4"]: !color,
                })}
            />
        </div>
    );
}

interface ProgressBarProps {
    percentage?: number;
    color?: string;
}
