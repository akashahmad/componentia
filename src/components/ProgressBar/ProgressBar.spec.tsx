import * as React from "react";

import { render } from "@testing-library/react";
import { ProgressBar } from "./ProgressBar";

test("ProgressBar component is rendering properly", async () => {
    const { container } = render(<ProgressBar />);

    expect(container).toMatchSnapshot();
});
