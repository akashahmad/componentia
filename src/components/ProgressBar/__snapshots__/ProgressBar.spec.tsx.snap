// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar component is rendering properly 1`] = `
<div>
  <div
    class="bg-grey-10 w-full h-[7px] rounded-full overflow-hidden inline-block"
  >
    <span
      class="inline-block h-[7px] absolute rounded-full bg-teal-light-4"
      style="width: 50%;"
    />
  </div>
</div>
`;
